parlai/core/utils.py

```diff
@@ -252,14 +246,26 @@ def parse(txt, splitSentences):
         elif len(history['labels']) > 0:
             r = history['labels'][0]
             history['dialog'].extend(parse(r, splitSentences))
-    if 'text' in observation:
-        history['dialog'].extend(parse(observation['text'], splitSentences))
-
-    history['episode_done'] = observation['episode_done']
-    if 'labels' in observation:
-        history['labels'] = observation['labels']
-    elif 'eval_labels' in observation:
-        history['labels'] = observation['eval_labels']
+    obs = observation
+    if 'text' in obs:
+        if useStartEndIndices:
+            if '\n' in obs['text']:
+                split = obs['text'].split('\n')
+                split[-1] = dict.end_token + ' ' + split[-1]
+                obs['text'] = '\n'.join(split)
+            else:
+                obs['text'] = dict.end_token + ' ' + obs['text']
```

Comment:
hmm..did you get my comment on this? seems lost
i dont get it.. how is splitting the 'text' in obs related to the people talking ..
and i think it should be based on id field